src/foam/lib/parse/ErrorReportingNodePStream.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.parse;
+
+/**
+ * This is a PStream decorator that listens for errors when parsing and reports them
+ * to the root ErrorReportingPStream
+ */
+public class ErrorReportingNodePStream
+    extends ProxyPStream
+{
+  protected int pos;
+  protected ErrorReportingPStream root;
```

Comment:
Should be consistent with instance variable naming. The FOAM convention is to have instance vars end with '_', but I don't mind if people don't, but in this case you mix both styles. So pick one and use it throughout a file (I prefer using _'s.)

Reply:
Added the underscore suffix.